linera-service/src/linera.rs

```diff
@@ -2281,28 +2303,36 @@ async fn run(options: ClientOptions) -> Result<(), anyhow::Error> {
                     panic!("The local test network must have at least one shard per validator.");
                 }
 
+                let (shutdown_sender, mut shutdown_receiver) = mpsc::channel(1);
+                tokio::spawn(handle_signals(shutdown_sender));
                 #[cfg(feature = "kubernetes")]
-                if *kubernetes {
-                    let config = LocalKubernetesNetConfig {
-                        network: Network::Grpc,
-                        testing_prng_seed: *testing_prng_seed,
-                        num_initial_validators: *validators,
-                        num_shards: *shards,
-                        binaries_dir: binaries_dir.clone(),
-                    };
-                    let (net, client1) = config.instantiate().await?;
-                    Ok(net_up(extra_wallets, net, client1).await?)
-                } else {
-                    let config = LocalNetConfig {
-                        network: Network::Grpc,
-                        database: Database::RocksDb,
-                        testing_prng_seed: *testing_prng_seed,
-                        table_name: table_name.to_string(),
-                        num_initial_validators: *validators,
-                        num_shards: *shards,
-                    };
-                    let (net, client1) = config.instantiate().await?;
-                    Ok(net_up(extra_wallets, net, client1).await?)
+                {
+                    if *kubernetes {
+                        let config = LocalKubernetesNetConfig {
+                            network: Network::Grpc,
+                            testing_prng_seed: *testing_prng_seed,
+                            num_initial_validators: *validators,
+                            num_shards: *shards,
+                            binaries_dir: binaries_dir.clone(),
+                        };
+                        let (mut net, client1) = config.instantiate().await?;
+                        let result = Ok(net_up(extra_wallets, &mut net, client1).await?);
+                        listen_for_signals(&mut shutdown_receiver, &mut net).await?;
+                        result
+                    } else {
```

Comment:
It's a bit sad to duplicate the content of the `else` branch. How about starting with something like this?
```
#[cfg(feature = "kubernetes")]
let (kubernetes, binaries_dirs) = (*kubernetes, binaries_dirs.clone());
#[cfg(not(feature = "kubernetes"))]
let (kubernetes, binaries_dirs) = (*kubernetes, PathBuf::default());
```

Reply:
Ah, true, good idea! Will do